app/src/main/java/com/example/servicesapp/MainActivity.kt

```diff
@@ -63,6 +64,12 @@ class MainActivity : AppCompatActivity(), PushServiceObserver, ProviderInstaller
             REQUEST_CODE_PERMISSION_GPS
         )
         PushServiceManagerProxy.addObserver(this)
+        PushServiceManagerProxy.obtainPushToken(object : IToken {
```

Comment:
This part confuses me a bit because of the name.
Since this is a system sender-receiver, names should be something like
setActivePushTokenListener, and instead of IToken it should be ActivePushTokenListner.
This will increase understanding, because now it seems like i must to put token instead of using it.

And in general it's strange to have listener for get current token, but there is no way to do this differently, since that thread must be used in hms service.

Reply:
Renamed